tornado/locks.py

```diff
@@ -0,0 +1,62 @@
+__all__ = ['Condition']
+
+import collections
+
+from tornado import gen, ioloop
+from tornado.concurrent import Future
+
+
+class Condition(object):
+    """A condition allows one or more coroutines to wait until notified.
+
+    Like a standard Condition_, but does not need an underlying lock that
+    is acquired and released.
+
+    .. _Condition: http://docs.python.org/library/threading.html#threading.Condition
```

Comment:
We have intersphinx available so you can refer to the standard library by just putting `threading.Condition` in backquotes.